[APM] Design - Adding main navigation bar to Home

formgeist opened the pull request:
**WIP - needs clean up and functionality implemented before merge**

# Summary

Related to https://github.com/elastic/observability-dev/issues/150

Design to combine the APM overview page links Services and Traces with the new Settings page link.

<img width="1792" alt="Screenshot 2019-06-27 at 22 26 58" src="https://user-images.githubusercontent.com/4104278/60298626-35fe1c00-992b-11e9-9636-49285e1e18a9.png">

### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] ~~This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)~~
- [ ] ~~Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~~
- [ ] ~~[Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~~
- [ ] ~~[Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~~
- [ ] ~~This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~~

### For maintainers

- [ ] ~~This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~
- [ ] ~~This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)~~

sorenlouv committed:
[APM] Central configuration management

ogupte committed:
[APM] Agent remote configuration UI
  - Creates apmCm index mapping on plugin initialization.
  - Links from home page to settings page
  - Polishes settings page to match design

formgeist committed:
Copy edits on ListSettings page

formgeist committed:
Updated snapshot

ogupte committed:
- Updates setting config flyout to match design.
- Adds ability to update/edit existing setting value
- Adds delete functionality to edit settings flyout

formgeist committed:
Adding main navigation bar for Home

elasticmachine commented:
Pinging @elastic/apm-ui

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/4328/)

formgeist committed:
Hide general padding

Needs fix for all views

formgeist committed:
Fixes padding

Since disabled in general, applying it to the view.

formgeist committed:
Added variables

formgeist committed:
Adds header wrapper

formgeist committed:
Style changes to navigation bar

formgeist commented:
@hbharding Made some style changes based on your comments in https://github.com/elastic/observability-dev/issues/150#issuecomment-506415603 - it's ugly and throwaway code, but gives us a sense of how it might look on the page if we had enclosed the header and navigation in the same top element with a different background.

formgeist commented:
Just fyi, those other tabs below the header - I can't get rid of those in a sensible way, because it's tied to the table. Needs engineering help detaching it.

hbharding commented:
I really like where this is going! 2 thoughts:
- The white background helps the search bar and other inputs standout more due to slightly higher contrast. In current Kibana, I think they can be hard to see against the grey background.
- If we drop the header "Services" below the search bar / outside of the white container, I wonder if we can reduce the white container into a single row of inputs, i.e. a "search bar". If we follow SIEM's approach, the search bar would transition to position:fixed as you scroll. Also - I think dropping the header below the search bar will make it more intuitive to the user that the search applies across tabs.

formgeist commented:
> If we drop the header "Services" below the search bar / outside of the white container, I wonder if we can reduce the white container into a single row of inputs, i.e. a "search bar". If we follow SIEM's approach, the search bar would transition to position:fixed as you scroll. Also - I think dropping the header below the search bar will make it more intuitive to the user that the search applies across tabs.

Yeah, so the reason we already moved the date picker up was because we were anticipating more global filter selectors like the one for Environments in future updates. Whenever we add a new selector, we're reducing the size of the query bar. When using the query bar, you're more than likely to have several queries entered which will be hidden if the query bar is too small. Perhaps we should take this idea further and make mocks for each of the solutions so we can compare and make them consistent.

cc @katrin-freihofner FYI

hbharding commented:
> Perhaps we should take this idea further and make mocks for each of the solutions so we can compare and make them consistent.

Happy to throw some ideas together!

Github / process question: Should we make a separate issue for the query bar / input area to decouple it from the top level navigation changes? Or, should we do the exploration as part of this issue? If so, do we assign all 3 of us to this issue?

formgeist commented:
@hbharding Yeah, it might be wise to separate the layout of the query bar and filter options to an APM UI design issue in Kibana, tagged `Team:apm`. And tbh, this PR was a little experiment, so I'd also prefer we make due process and create an issue for the header design perhaps in observability-dev repo. Would you mind setting up those two issues?

formgeist commented:
Alas, I've already created a general design issue for figuring out the navigation pattern across Obs solutions, so I reckon we can continue there. https://github.com/elastic/observability-dev/issues/150

I've also created a separate issue for the APM UI header input and "global" filters layout https://github.com/elastic/kibana/issues/40236
